packages/angular_devkit/build_angular/test/browser/aot_spec_large.ts

```diff
@@ -39,4 +39,28 @@ describe('Browser Builder AOT', () => {
 
     await run.stop();
   });
+
+  it('shows warnings for component styles', async () => {
+    const overrides = {
+      aot: true,
+      optimization: true,
+    };
+
+    host.writeMultipleFiles({
+      'src/app/app.component.css': `
+        .foo { color: white; padding: 1px; };
+        .buz { color: white; padding: 2px; };
+      `,
+    });
+
+    const logger = new logging.Logger('');
+    const logs: string[] = [];
+    logger.subscribe(e => logs.push(e.message));
+
+    const run = await architect.scheduleTarget(targetSpec, overrides, { logger });
+    const output = await run.result;
+    expect(output.success).toBe(true);
+    expect(logs.join()).toContain('WARNING in Invalid selector');
```

Comment:
I don't understand what this warning means. Is it related to bundle budgets?

Reply:
No it's related to the other fix which is required for the bundle budgets

https://github.com/angular/angular-cli/pull/15127/commits/39f1be02beb7391bfbb34cd4c342d97c94d8fb1c

Reply:
Discussed with Alan a bit more. In this example the semicolons are actually causing the `.buz` class to be dropped by the optimizer, but the user is not warned. When this goes in users with this sort of warning will start seeing them.